buildenv/jenkins/JenkinsfileBase

```diff
@@ -401,6 +401,8 @@ def setup() {
 		if (params.SDK_RESOURCE == 'nightly' && params.CUSTOMIZED_SDK_URL) {
 			// remove single quote to allow variables to be set in CUSTOMIZED_SDK_URL
 			CUSTOMIZED_SDK_URL_OPTION = "-c ${params.CUSTOMIZED_SDK_URL}"
+		} else if (params.SDK_RESOURCE != 'customized') {
+			CUSTOMIZED_SDK_URL_OPTION = ""
```

Comment:
Here, we assume if user provides SDK_RESOURCE != 'customized' and CUSTOMIZED_SDK_URL, we will ignore CUSTOMIZED_SDK_URL value and proceed with SDK_RESOURCE value only.
IMO, we should just fail in this case.

I would like to rewrite the whole logic to be more readable:

```
if (params.CUSTOMIZED_SDK_URL) {
	if (params.SDK_RESOURCE == 'nightly') {
		// remove single quote to allow variables to be set in CUSTOMIZED_SDK_URL
        	CUSTOMIZED_SDK_URL_OPTION = "-c ${params.CUSTOMIZED_SDK_URL}"
	} else if (params.SDK_RESOURCE == 'customized' || !params.SDK_RESOURCE) {
		SDK_RESOURCE = "customized"
		CUSTOMIZED_SDK_URL_OPTION = "-c '${params.CUSTOMIZED_SDK_URL}'"
                … // RI_JDK code
        } else {
		error("SDK_RESOURCE: ${params.SDK_RESOURCE} and CUSTOMIZED_SDK_URL: ${params.CUSTOMIZED_SDK_URL} combo is not supported!)
        }
} else {
        if (params.SDK_RESOURCE == 'customized') {
		error("SDK_RESOURCE: ${params.SDK_RESOURCE}, please provide CUSTOMIZED_SDK_URL")
        }
}
```

Reply:
This is a more strict user story. Updated.